### ISSUE TITLE
Date filter giving off by one dates

### ISSUE THREAD

**Issue by scottcain** [cancogen-virus-seq/portal#162]
This issue was initially raised by @nithujohn, as the result of a Compute Canada ticket (Ticket#0147824)

When I try to filter data by collection date and, for example, select the start and end date to Dec 10, I get results for Dec 11. Please see the attached screenshot.

<img width="959" alt="Screen Shot 2021-12-30 at 12 52 56 PM" src="https://user-images.githubusercontent.com/274751/147787445-26a26953-1449-4794-a1dc-f1feee71f2aa.png">

**Comment by scottcain** [comment]
Guessing @jaserud is a good person to assign this to--if not, please pass along to the right person.

**Comment by jaserud** [comment]
AFAIK Arranger provides the filtering here so I think Jon or Justin is better suited for this one. Reassigned to them.

**Comment by caravinci** [comment]
### As mentioned in slack, results of my research into this issue:
Based on the response from the Arranger server, there seems to be a format mismatch in the results for `sample_collection_date`, as it is expected to either be formatted in ISO or milliseconds. See here.
<img width="309" alt="image" src="https://user-images.githubusercontent.com/2107110/148120114-fcba0bf7-63e0-408f-a9db-abe93e6e11d1.png">

will amend this comment with further insights

**Comment by caravinci** [comment]
On further study, the data seems to be fed to Elastic search inconsistently, as evidenced here:
<img width="340" alt="image" src="https://user-images.githubusercontent.com/2107110/148249894-0eb32e57-67ab-4cbe-b277-29a9df0f16a8.png">
<img width="494" alt="image" src="https://user-images.githubusercontent.com/2107110/148249971-76c6c7cb-1277-41d7-9ed6-7f21d0719e5f.png">


This may be a result of something wrong in Score, or even Maestro (not sure where the issue comes from exactly). Will continue doing research with @jaserud to identify and correct it.

**Comment by caravinci** [comment]
After a few tests with @jaserud we have determined this issue has three parts:
- Arranger UI (e.g. the exploration table) is displaying the UTC date strings given by ES directly "as is", effectively shifting some of the dates while querying without displaying the timezone, which makes the issue hard to identify. 
This is easily resolved on render, by converting from UTC to the user's local timezone, which is likely the users' assumption.
- This same problem will be noticed in the download bundles, as their displayed dates either have to be shifted as well for consistency, or at least clarified as being displayed in UTC. (prompt for @scottcain @akachru-github for decision on how to proceed). 
- This uncovered a bug in the back-end, while uploading data: the dates are being submitted in their own local times, but are assumed as UTC, which is effectively corrupting the information. We discussed a possible solution, to be applied in Muse (possibly needing a Song migration as well)... this part needs to be thought+discussed in more detail, as it is not a trivial implementation.
